src/sagemaker/tensorflow/estimator.py

```diff
@@ -362,18 +410,39 @@ def hyperparameters(self):
         """Return hyperparameters used by your custom TensorFlow code during model training."""
         hyperparameters = super(TensorFlow, self).hyperparameters()
 
-        if not self.checkpoint_path:
-            local_code = get_config_value('local.local_code', self.sagemaker_session.config)
-            if self.sagemaker_session.local_mode and local_code:
-                self.checkpoint_path = '/opt/ml/shared/checkpoints'
-            else:
-                self.checkpoint_path = os.path.join(self.output_path,
-                                                    self._current_job_name, 'checkpoints')
+        self.checkpoint_path = self.checkpoint_path or self._default_s3_path('checkpoints')
```

Comment:
Is this parameter not used in script mode? Let's not set it if that is the case to avoid future breaking changes.

Reply:
this need to be set for the framework mode containers for now. We will need to remove them in the future.